content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,322 @@
+
+1. [**Create a new repository on GitHub**](https://happygitwithr.com/new-github-first.html#make-a-repo-on-github-2) for your work.
+
+    * The repository name you choose will also end up being the name of your package, so choose wisely! For the examples in this guide, we'll go with "tutorialpackage".
+
+1. **Copy the repository URL** to your clipboard.
+
+    * Do this by clicking the green Clone or Download button.
+    * Copy the HTTPS URL (looks like: `https://github.com/{yourname}/{yourrepo}.git`).
+
+    * **Or**, copy the SSH URL if you chose to set up SSH keys (looks like: `git@github.com:{yourname}/{yourrepo}.git`).
+
+1. **Open RStudio** and [create a new RStudio Project via git clone](https://happygitwithr.com/new-github-first.html#new-rstudio-project-via-git-clone).
+
+    * Do this by clicking File > New Project > Version Control > Git.
+
+    * Paste the copied URL.
+
+    * Be intentional about where you tell RStudio to create this new Project.
+
+1. **Click Create Project.**
+
+    * Your R Project is now connected to GitHub!
+
+
+
+### Add files and directories for a package
+
+Now we'll create the infrastructure to turn this R project into a *package*. A handful of new files need to be created, but luckily we can outsource this heavy lifting to the `create_package()` function from the [usethis](https://usethis.r-lib.org/) package.
+
+1.  **Create the necessary package infrastructure** when you run:
+
+    ```{r, eval = FALSE}
+    usethis::create_package(getwd())
+    ```
+
+2. The console output will ask if you’d like to overwrite the pre-existing R project. __Select *No*__.
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-overwrite.png")
+    ```
+
+3. A second session of RStudio will open – this session has a *Build* tab in the pane which also has the tabs Environment, History, etc. This tab is specific for building packages, and we will use it later. **You can close the first RStudio instance**.
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-build.png")
+    ```
+
+    * Your project directory should end up looking something like this:
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-directory.png")
+    ```
+
+
+
+### Add files and directories for a tutorial
+
+Now that the package structure is in place, it's time to add tutorial content. Again, the usethis package will do the dirty work for us to create the folders and files we need for a learnr tutorial within our package.
+
+1. **Create the learnr tutorial** and open it for editing when you run `usethis::use_tutorial("<tutorial-name>", "<Tutorial Title>")`:
+
+    ```{r, eval = FALSE}
+    usethis::use_tutorial("lesson1", "Your First Lesson", open = interactive())
+    ```
+
+    * The first argument is the basename for the learnr R Markdown file you are creating. You do not need the `.Rmd` file extension here.
```

Comment:
Is "basename" a technical term? I've not used it before. If it's commonly known, it's fine. Otherwise I'd just call the "filename without the `.Rmd` extension".